### PR TITLE
reports for edtf dates with x or X

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made? 🤔

Part of #4143, this is the code for the requested reports:  edtf dates that have lower case x  and another report for upper case X.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: app/reports/edtf_lower_x_dates.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      new(dro:).report
+    end
+  end
+
+  def initialize(dro:)
+    @dro = dro
+  end
+
+  REGEX_OPTS = Regexp::IGNORECASE | Regexp::MULTILINE
+  DECADE_4CHAR_REGEX = Regexp.new('(^|\D)\d{3}[u\-?x]($|\D)', REGEX_OPTS)
```
Comment (mjgiarlo) [inline app/reports/edtf_lower_x_dates.rb:19]: Hmm, where is this used?

File: app/reports/edtf_upper_x_dates.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      new(dro:).report
+    end
+  end
+
+  def initialize(dro:)
+    @dro = dro
+  end
+
+  REGEX_OPTS = Regexp::IGNORECASE | Regexp::MULTILINE
+  DECADE_4CHAR_REGEX = Regexp.new('(^|\D)\d{3}[u\-?x]($|\D)', REGEX_OPTS)
```
Comment (mjgiarlo) [inline app/reports/edtf_upper_x_dates.rb:19]: Same question.

**Review by mjgiarlo — COMMENTED** [review]
Have you run the report yet? You may want to run it in prod and make sure it does what you expect and then mark this PR as ready for review or unheld or whatever. (At least, I've found this useful in the past as I inevitably get stuff wrong that I don't notice until I run it...)

File: app/reports/edtf_lower_x_dates.rb (hunk 1 above)
Comment (ndushay) [reply app/reports/edtf_lower_x_dates.rb:19]: whoops - cruft.

**Comment by ndushay** [comment]
I have run a report with '1' substituted for 'x' with `.limit(10)` and got output, so I'm confident these work.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**